### PR TITLE
MM-9929 Technical description of SAML sync with AD/LDAP

### PR DESCRIPTION
**Pull request by jasonblais**
https://mattermost.atlassian.net/browse/MM-9929

Credit to @coreyhulen

### CODE REVIEW

File: source/deployment/sso-saml-ldapsync.rst [hunk 1]
```diff
@@ -29,3 +29,20 @@ Once the synchronization with AD/LDAP is enabled, user attributes are synchroniz
     
     If a user is deactivated from SAML, their session won't expire until they're deactivated from AD/LDAP. However, they won't be able to log back in to Mattermost.
  
+Technical description of SAML synchronization with AD/LDAP
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When enabled, SAML synchronization with AD/LDAP occurs in phases:
+
+1. Get all the current LDAP users from the Mattermost database who have ``Users.AuthService`` set to ``ldap``. This is a SQL query issued against the Mattermost database.
```
Comment (jasonblais) [inline source/deployment/sso-saml-ldapsync.rst]: Can we add what the SQL query is?

File: source/deployment/sso-saml-ldapsync.rst [hunk 2]
```diff
@@ -29,3 +29,20 @@ Once the synchronization with AD/LDAP is enabled, user attributes are synchroniz
...
+2. Get all the current SAML users from the Mattermost database who have ``Users.AuthService`` set to ``saml``. This is a SQL query issued against the Mattermost database.
```
Comment (jasonblais) [inline source/deployment/sso-saml-ldapsync.rst]: Can we add what the SQL query is?

File: source/deployment/sso-saml-ldapsync.rst [hunk 3]
```diff
@@ -29,3 +29,20 @@ Once the synchronization with AD/LDAP is enabled, user attributes are synchroniz
...
+3. Get all the current LDAP users from the LDAP server as defined by ``LdapSettings.UserFilter``. This is an LDAP query issued against the LDAP server. Users are retrieved in batches as defined by ``LdapSettings.MaxPageSize``.
```
Comment (jasonblais) [inline source/deployment/sso-saml-ldapsync.rst]: Please review steps 3 - 5 for accuracy. Would it be worth adding the LDAP query to step 3?

Comment (coreyhulen) [reply source/deployment/sso-saml-ldapsync.rst]: Maybe add a link to most/mattermost-server/blob/master/scripts/ldap-check.sh for the query

File: source/deployment/sso-saml-ldapsync.rst (hunk 1 above)
Comment (coreyhulen) [reply source/deployment/sso-saml-ldapsync.rst]: `SELECT * FROM Users WHERE AuthService = :AuthService"`

File: source/deployment/sso-saml-ldapsync.rst (hunk 2 above)
Comment (coreyhulen) [reply source/deployment/sso-saml-ldapsync.rst]: same query as above `SELECT * FROM Users WHERE AuthService = :AuthService"`

File: source/deployment/sso-saml-ldapsync.rst [hunk 4]
```diff
@@ -29,3 +29,20 @@ Once the synchronization with AD/LDAP is enabled, user attributes are synchroniz
...
+4. Update LDAP attributes. For each existing Mattermost user retrieved in step 1, attempt to find a match against the list of LDAP users from step 3. To find matches, ``Users.AuthService`` field of the Mattermost user is compared against the ``LdapSettings.IdAttribute`` LDAP setting.
```
Comment (coreyhulen) [inline source/deployment/sso-saml-ldapsync.rst]: oops...in step 4 `Users.AuthService` should be `Users.AuthData`

**Comment by jasonblais** [comment]
Thanks! @coreyhulen updates made

**Pull request merged**